libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -235,6 +323,1154 @@ static size_t encodeString( uint8_t * pDestination,
+    connectInfo.clientIdentifierLength = CLIENT_IDENTIFIER_LENGTH;
+    status = MQTT_GetConnectPacketSize( &connectInfo, NULL, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    /* Verify good case */
+    memset( ( void * ) &connectInfo, 0x0, sizeof( connectInfo ) );
+    connectInfo.cleanSession = true;
+    connectInfo.pClientIdentifier = "TEST";
+    connectInfo.clientIdentifierLength = 4;
+    status = MQTT_GetConnectPacketSize( &connectInfo, NULL, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    /* Make sure remaining size returned is 16. */
+    TEST_ASSERT_EQUAL_INT( 16, remainingLength );
+    /* Make sure packet size is 18. */
+    TEST_ASSERT_EQUAL_INT( 18, packetSize );
+
+    /* With will. These parameters will cause the packet to be
+     * 4 + 2 + 8 + 2 = 16 bytes larger. */
+    willInfo.pTopicName = "test";
+    willInfo.topicNameLength = 4;
+    willInfo.pPayload = "testload";
+    willInfo.payloadLength = 8;
+    status = MQTT_GetConnectPacketSize( &connectInfo, &willInfo, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    /* Make sure remaining size returned is 32 = 16 + 16. */
+    TEST_ASSERT_EQUAL_INT( 32, remainingLength );
+    /* Make sure packet size is 34 = 18 + 16. */
+    TEST_ASSERT_EQUAL_INT( 34, packetSize );
+}
+
+/**
+ * @brief Tests that MQTT_SerializeConnect works as intended.
+ */
+void test_MQTT_SerializeConnect( void )
+{
+    MQTTConnectInfo_t connectInfo;
+    MQTTPublishInfo_t willInfo;
+    size_t remainingLength = 0;
+    uint8_t buffer[ 70 ];
+    size_t bufferSize = sizeof( buffer );
+    size_t packetSize = bufferSize;
+    MQTTStatus_t status = MQTTSuccess;
+    MQTTFixedBuffer_t fixedBuffer = { .pBuffer = buffer, .size = bufferSize };
+
+    /* Verify bad parameter errors. */
+    status = MQTT_SerializeConnect( NULL, &willInfo, remainingLength, &fixedBuffer );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+    status = MQTT_SerializeConnect( &connectInfo, &willInfo, remainingLength, NULL );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    memset( ( void * ) &connectInfo, 0x0, sizeof( connectInfo ) );
+    status = MQTT_SerializeConnect( &connectInfo, NULL, 120, &fixedBuffer );
+    TEST_ASSERT_EQUAL_INT( MQTTNoMemory, status );
+
+    /* Connect packet too large. */
+    memset( ( void * ) &connectInfo, 0x0, sizeof( connectInfo ) );
+    connectInfo.pClientIdentifier = CLIENT_IDENTIFIER;
+    connectInfo.clientIdentifierLength = UINT16_MAX;
+    connectInfo.pPassword = "";
+    connectInfo.passwordLength = UINT16_MAX;
+    connectInfo.pUserName = "";
+    connectInfo.userNameLength = UINT16_MAX;
+    willInfo.pTopicName = TEST_TOPIC_NAME;
+    willInfo.topicNameLength = UINT16_MAX;
+    willInfo.payloadLength = UINT16_MAX + 2;
+    status = MQTT_GetConnectPacketSize( &connectInfo, &willInfo, &remainingLength, &packetSize );
```

Comment:
This is unit tests for MQTT_GetConnectPacketSize as mentioned in the earlier comment.

Reply:
Good catch, it seems this was misplaced in v4_beta and copied over to here. I'll move the test and delete it here